plugin/plugin.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 the Drone Authors. All rights reserved.
+// Use of this source code is governed by the Blue Oak Model License
+// that can be found in the LICENSE file.
+
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/georgeJobs/go-antpathmatcher"
+	"github.com/sirupsen/logrus"
+)
+
+// Args provides plugin execution arguments.
+type Args struct {
+	Pipeline
+
+	// Level defines the plugin log level.
+	Level string `envconfig:"PLUGIN_LOG_LEVEL"`
+
+	// Ant style pattern Glob pattern to search for files. (required)
+	Filter string `envconfig:"PLUGIN_FILTER"`
```

Comment:
`glob` argument is used in the Jenkins findFiles plugin and most of the plugins like tar, untar, zip, and gzip. The new `plugins/archive` will also use glob as input. Let’s keep the same arguments for consistency and clarity.

Reply:
Done